gadget/src/lib.rs

```diff
@@ -34,12 +38,55 @@ pub async fn ignite_lc_relayer(ctx: LightClientRelayerContext) -> anyhow::Result
 			},
 		};
 		let api_client = Arc::new(api_client);
-		let pair = std::fs::read_to_string(&ctx.lc_init_config.path_to_signer_secret_key)
-			.expect("failed to read secret key");
-		let pair = subxt::ext::sp_core::sr25519::Pair::from_string(&pair, None);
+		// Read path first will change to string later for parsing
+		let path = Path::new(&ctx.lc_init_config.path_to_signer_secret_key);
+		// Check if the file exists, its permissions, and if it's empty
+        let pair = if path.exists() {
+			match fs::metadata(path) {
+				Ok(metadata) => {
+					let permissions = metadata.permissions();
+					if permissions.mode() & 0o777 == 0o600 {
+						match fs::read_to_string(path) {
+							Ok(content) if !content.trim().is_empty() => {
+								match parse_suri(&content) {
+									Some(s) => subxt::ext::sp_core::sr25519::Pair::from_string(&s, None).ok(),
+									None => {
+										tracing::info!("Invalid SURI format in the file.");
+										None
+									}
+								}
+							}
+							Ok(_) => {
+								tracing::info!("Secret key file is empty");
+								None
+							}
+							Err(e) => {
+								tracing::info!("Failed to read the secret key file: {:?}", e);
+								None
+							}
+						}
+					} else {
+						tracing::info!("Invalid file permissions: {:o}. Required permissions: 0600", permissions.mode());
+						None
+					}
+				},
+				Err(e) => {
+					tracing::info!("Failed to get metadata for the secret key file: {:?}", e);
+					None
+				}
+			}
+		} else {
+			tracing::info!("Secret key file does not exist at the specified path: {:?}", path);
+			None
+		};
 		let network = ctx.lc_relay_config.ethereum_network.as_typed_chain_id();
-		let mut eth_pallet = if let Ok(pair) = pair {
-			tracing::info!(target: "relay", "=== Initializing client with signer ===");
+		let mut eth_pallet = if let Some(pair) = pair {
+			// Substrate default addr prefix
+			let pub_addr = pair.public().to_ss58check();
+			// GGX Prefix
+			let custom_prefix = Ss58AddressFormat::custom(8888);
```

Comment:
we have different prefix for brooklyn and sydney, and mainnet. so it should be configurebale

Reply:
Ah yes good point will add that as an option, any suggestions where youd prefer or an idea for its configuration?